src/main/java/com/epam/test_generator/services/JiraService.java

```diff
@@ -0,0 +1,343 @@
+     *
+     * @return list projects
+     */
+    public List<JiraProject> getNonexistentJiraProjects() throws JiraException {
+        return jiraProjectDAO.getAllProjects().stream()
+            .filter(project -> projectDAO.findByJiraKey(project.getJiraKey()) == null)
+            .collect(Collectors.toList());
+    }
+
+    /**
+     * Synchronisation method from Jira This method receives all related and opened stories and sub
+     * stories from Jira and update its in BDD. <p> In case where issue has closed in Jira - the
+     * changes not present in the response. By this reason all existing cases or suits which not
+     * presents in response will be deleted from BBD.
+     */
+    public void syncFromJira() throws JiraException {
+        String projects = projectDAO.findAll().stream()
+            .filter(p -> p.getJiraKey() != null)
+            .map(Project::getJiraKey)
+            .collect(Collectors.joining(","));
+
+        String query = String.format(
+            "project in (%s) AND status in (Open, \"In Progress\", Reopened, Verified) AND type=story",
+            projects);
+
+        String subtaskQuery = String.format(
+            "project in (%s) AND status in (Open, \"In Progress\", Reopened, Verified) AND type=sub-task",
+            projects);
+
+        List<JiraStory> jiraStories = jiraStoryDAO.getJiraStoriesByFilter(query);
+        List<JiraSubTask> jiraSubTasks = jiraSubStoryDAO.getJiraSubtoriesByFilter(subtaskQuery);
+
+        Map<String, Suit> suits = suitDAO.findAll().stream().filter(suit ->
+            jiraStories.stream().filter(jiraStory -> jiraStory.getJiraKey() != null).noneMatch(jiraStory -> jiraStory.getJiraKey().equals(suit.getJiraKey())))
+            .collect(Collectors.toMap(Suit::getJiraKey, suit -> suit));
+
+        Map<String, Case> cases = caseDAO.findAll().stream().filter(caze ->
+            jiraSubTasks.stream().filter(jiraSubTask -> jiraSubTask.getJiraKey() != null).noneMatch(jiraSubTask -> jiraSubTask.getJiraKey().equals(caze.getJiraKey())))
+            .collect(Collectors.toMap(Case::getJiraKey, caze -> caze));
+
+
+
+        jiraStories.forEach(this::updateSuitFromJiraStory);
+
+        jiraSubTasks.stream()
+            .filter(subTask -> suitDAO.findByJiraKey(subTask.getJiraParentKey()) != null)
+            .forEach(subTask -> {
+                Case aCase = caseDAO.findByJiraKey(subTask.getJiraKey());
+                if (aCase == null) {
+                    createCaseFromJiraSubTask(subTask);
+                } else {
+                    updateSubTaskFromJiraSubTask(subTask);
+                }
+            });
+
+        suits.forEach((jiraKey, suit) -> {
+            projectDAO.findByJiraKey(suit.getJiraProjectKey()).getSuits().remove(suit);
+            suitDAO.delete(suit);
+        });
+        cases.forEach((jiraKey, caze) -> {
+            suitDAO.findByJiraKey(caze.getJiraParentKey()).getCases().remove(caze);
+            caseDAO.delete(caze);
+        });
+
+    }
```

Comment:
It is a very long method.
Another developer will spend time scrolling it and understand what happens in this fragment.
Read Fowler's article about function length here https://martinfowler.com/bliki/FunctionLength.html
May be you could extract something in private methods?